contracts/contracts/structs/CrossNet.sol

```diff
@@ -29,6 +29,31 @@ struct BottomUpCheckpoint {
     uint64 nextConfigurationNumber;
     /// @dev Batch of messages to execute.
     IpcEnvelope[] msgs;
+    /// @dev A commitment to the summary of our chain activity since the previous checkpoint and this one.
+    bytes32 summary;
+    /// @dev Summaries relayed upwards from descendants of this subnet.
+    /// NOTE: Not merkelized to keep it simple, but we will merkelize later to scale better.
+    RelayedSummary[] relayedSummaries;
+}
+
+struct ActivitySummary {
+    /// @dev The block range the activity summary spans; these are the local heights of the start and the end, inclusive.
+    uint256[2] blockRange;
+    /// @dev The validators whose activity we're reporting about.
+    address[] validators;
+    /// @dev The number of blocks committed by each validator in the position they appear in the validators array.
+    /// If there is a configuration change applied at this checkpoint, this carries information about the _old_ validator set.
+    uint64[] blocksCommitted;
```

Comment:
Could we have a metadata bytes typed member here to allow to pass additional info?

Reply:
Nice suggestion @JulissaDantes. @raulk, in particular, we have additional information we'd like to propagate up the hierarchy. I know we had talked about specific "zones" for this type of information in the checkpoints at one point. What is your thinking here at this stage?

Reply:
At least for solidity, there is no generic, then perhaps only bytes array? To let the subnet/fendermint and parent IValidatorRewarder handle the bytes deserialization + distribution. What the framework does is just checking:
- Validator has claimed before?
- Does the claim actually exists?
- The lifecycle of the validator reward

Reply:
Yes, this is the direction I wanted to move things in. This was the idea:
1. We add a region in checkpoints to carry extensible subnet activity reports in the form of `{ namespace/type/discriminator: bytes, payload: bytes }[]` -- we can call this checkpoint extensions.
2. IPC ships with a set of default subnet activity reporters as extensions, e.g. the block production one here.
3. Users can plug in custom subnet activity reporters to, e.g. report on quality of service, retrieval rate, scoring, etc.
4. Ideally the reporters are Wasm actors, so they can compose easily and be upgraded through governance in the future. However, to truly make this design client/node-agnostic, users would likely need to add custom syscalls to access the metadata actors will want to report on.
5. We'd need a master actor to orchestrate "hooks". The `extend_checkpoint` hook would be called by the IPC node, injecting the raw Checkpoint and expecting an array of extension fragments in return. The client would assemble them into the Checkpoint and submit that on chain for local quorum signature.

@carsonfarmer @JulissaDantes In my head we could defer this feature, but it sounds like it would be beneficial to you guys to have it now, right?

If so, I think we can simplify the above on several fronts to try to meet the time constraints while delivering the feature. How about:

1. We don't bother with developing the checkpoint extensions as Wasm-land actors for now; we just add well-bounded hook callbacks in Fendermint with the intention to migrate them to actors in the future.
2. We have the node orchestrate the hooks instead of a master actor.

---

Open points:

1. Namespacing.
2. Parent execution.